src/main/java/org/kohsuke/groovy/sandbox/ScopeTrackingClassCodeExpressionTransformer.java

```diff
@@ -157,15 +157,9 @@ public void visitCatchStatement(CatchStatement statement) {
 
     @Override
     public void visitWhileLoop(WhileStatement loop) {
+        loop.setBooleanExpression((BooleanExpression) transform(loop.getBooleanExpression()));
         try (StackVariableSet scope = new StackVariableSet(this)) {
-            super.visitWhileLoop(loop);
-        }
-    }
-
-    @Override
-    public void visitClosureExpression(ClosureExpression expression) {
```

Comment:
I always thought it was somewhat unusual for us to have a method that visited an `Expression` here when essentially everything else we visit a statement. After the other changes in this class, you can change the body of the method to `throw new RuntimeException(expression);` and see that it never gets called anymore.